s3tests_boto3/functional/test_iam.py

```diff
@@ -2052,3 +2052,179 @@ def test_allow_deny_put_bucket_versioning_iam_policy_others():
     _delete_all_objects(s3_client_alt, bucket)
     response = s3_client_iam.delete_bucket(Bucket=bucket)
     eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Test Allow and Deny Put Object Version Tagging API using IAM policy for self')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_deny_put_object_version_tagging_iam_policy_self():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    obj_key = "iam1buk1obj1"
+
+    # Create bucket, enable versioning, upload object
+    bucket = get_new_bucket(client=s3_client_iam)
+    response = s3_client_iam.put_bucket_versioning(Bucket=bucket,
+                                                   VersioningConfiguration={"Status": "Enabled"})
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    response = s3_client_iam.get_bucket_versioning(Bucket=bucket)
+    eq(response['Status'], 'Enabled')
+    version_ids = []
+    for _ in range(2):
+        response = s3_client_iam.put_object(Bucket=bucket, Key=obj_key, Body='bar')
```

Comment:
Instead of for loop, we can also use list_object_versions. It returns dictionary of object versions

Reply:
This code is to create multiple versions.